pyquil/operator_estimation.py

```diff
@@ -807,9 +843,98 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
             obs_vals = coeff * np.prod(my_obs_strings, axis=1)
             obs_mean = np.mean(obs_vals)
             obs_var = np.var(obs_vals) / n_shots
+
+            if calibrate_readout:
+                # 4 Readout calibration
+                # 4.1 Prepare the +1 eigenstate for the out operator
+                calibr_prog = Program()
+                for q, op in setting.out_operator.operations_as_set():
+                    calibr_prog += _one_q_pauli_prep(label=op, index=0, qubit=q)
+                # 4.2 Measure the out operator in this state
+                for q, op in setting.out_operator.operations_as_set():
+                    calibr_prog += _local_pauli_eig_meas(op, q)
+                calibr_results = qc.run_and_measure(calibr_prog, n_shots)
+                # 4.3 Calculate expectation value
+                obs_calibr_strings = {q: 1 - 2 * calibr_results[q] for q in calibr_results}
```

Comment:
This is all duplicated code. Consider factoring out into a helper function

Reply:
Updated.